core/src/stored/stored.cc

```diff
@@ -254,26 +265,32 @@ int main(int argc, char* argv[])
   }
 
   if (export_config) {
-    my_config->DumpResources(PrintMessage, NULL);
+    my_config->DumpResources(PrintMessage, nullptr);
     goto bail_out;
   }
 
+  if (!CheckResources()) {
+    Jmsg((JobControlRecord*)NULL, M_ERROR_TERM, 0,
+         _("Please correct the configuration in %s\n"),
+         my_config->get_base_config_path().c_str());
+  }
+
   if (!foreground && !test_config) {
-    daemon_start();  /* become daemon */
+    pid_t forkpid = daemon_start();
+    if (forkpid) {
+      if (pidfile_path) { WritePidFile(pidfile_fd, pidfile_path, forkpid); }
```

Comment:
you could also just pass the fd and the path down to daemon_start() and let that one handle the details.
Less duplication as you don't need to repeat the code in all three daemons.
```suggestion
    daemon_start(pidfile_fd, pidfile_path);
```